Use new NumPy pinnings

jakirkham opened the pull request:
Update this feedstock to use the [new style of NumPy pinnings in conda-forge]( https://conda-forge.org/docs/meta.html#building-against-numpy ). This should cutdown on the CI resources this feedstock uses and allow the deployed packages to be used with older versions of NumPy if needed. Also re-render to take advantage of the new CircleCI matrix and eliminate the NumPy CI matrices (as they are unneeded with the new NumPy pinnings).

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.4.2

jakirkham committed:
Use the new NumPy pinning

Switches over to the new style of NumPy pinning by using the oldest
supported version of NumPy to build with and setting that as a minimum
requirement at run time.

jakirkham committed:
MNT: Re-rendered with conda-smithy 2.4.2

Now that we are using the new style of NumPy pinnings, re-render to do
away with the old NumPy CI matrices.

jakirkham committed:
Bump build number to 1

Rebuild after changing the NumPy pinnings.
